entgql/template/node.tmpl

```diff
@@ -41,92 +41,27 @@ import (
 
 // Noder wraps the basic Node method.
 type Noder interface {
+{{- if hasTemplate "gql_client_node" }}
 	Node(context.Context) (*Node, error)
-}
-
-// Node in the graph.
-type Node struct {
-	ID 	   {{ $idType }} `json:"id,omitempty"`// node id.
-	Type   string   `json:"type,omitempty"`   // node type.
-	Fields []*Field `json:"fields,omitempty"` // node fields.
-	Edges  []*Edge  `json:"edges,omitempty"`  // node edges.
-}
-
-// Field of a node.
-type Field struct {
-	Type  string `json:"type,omitempty"`  // field type.
-	Name  string `json:"name,omitempty"`  // field name (as in struct).
-	Value string `json:"value,omitempty"` // stringified value.
-}
-
-// Edges between two nodes.
-type Edge struct {
-	Type string   `json:"type,omitempty"` // edge type.
-	Name string   `json:"name,omitempty"` // edge name.
-	IDs  []{{ $idType }} `json:"ids,omitempty"`  // node ids (where this edge point to).
+{{- end }}
+	IsNode()
```

Comment:
Maybe, `IsNoder`? because the interface is called Noder.

Reply:
IsNode(), because the GQL type was `Node`.